torch/utils/cpp_extension.py

```diff
@@ -2268,20 +2454,48 @@ def _write_ninja_file_to_build_library(path,
     else:
         cuda_flags = None
 
+    if with_sycl:
+        sycl_common_cflags = ['-fsycl', '-fsycl-targets=spir64_gen,spir64']
+        sycl_cflags = cflags + sycl_common_cflags
+        sycl_cflags += extra_sycl_cflags
+        # "preview breaking changes" flags allows to support both C++ ABIs
+        sycl_cflags += ['-fpreview-breaking-changes', '-D__INTEL_PREVIEW_BREAKING_CHANGES']
+        if not any(flag.startswith('-sycl-std=') for flag in sycl_cflags):
+            sycl_cflags.append('-sycl-std=2020')
+        host_cxx = get_cxx_compiler()
+        host_cflags = cflags
+        # escaping quoted arguments to pass them thru icpx
+        host_cflags = [item.replace('\\"', '\\\\"') for item in host_cflags]
+        host_cflags = ' '.join(host_cflags)
+        # note that -fsycl-host-compiler-options will be quoted as needed
+        # with shlex.quote below
+        sycl_cflags += [f'-fsycl-host-compiler={host_cxx}', shlex.quote(f'-fsycl-host-compiler-options={host_cflags}')]
+        sycl_arch_list = os.environ.get('TORCH_XPU_ARCH_LIST', 'pvc,xe-lpg')
+        sycl_dlink_post_cflags = sycl_common_cflags + ['-fsycl-link', f'-Xs "-device {sycl_arch_list}"']
+    else:
+        sycl_cflags = None
+        sycl_dlink_post_cflags = None
+
     def object_file_path(source_file: str) -> str:
         # '/path/to/file.cpp' -> 'file'
         file_name = os.path.splitext(os.path.basename(source_file))[0]
         if _is_cuda_file(source_file) and with_cuda:
             # Use a different object filename in case a C++ and CUDA file have
             # the same filename but different extension (.cpp vs. .cu).
             target = f'{file_name}.cuda.o'
+        elif _is_sycl_file(source_file) and with_sycl:
+            target = f'{file_name}.sycl.o'
         else:
             target = f'{file_name}.o'
         return target
 
     objects = [object_file_path(src) for src in sources]
     ldflags = ([] if is_standalone else [SHARED_FLAG]) + extra_ldflags
 
+    if with_sycl:
+        # "sycl-preview" library allows to support both C++ ABIs
+        ldflags += ['-lsycl-preview']
```

Comment:
The Intel GPU development bundle has upgraded to the latest version.
```suggestion
        ldflags += ['-lsycl]
```